propagators/aws/xray/idgenerator.go

```diff
@@ -19,16 +19,37 @@ import (
 
 // IDGenerator is used for generating a new traceID and spanID.
 type IDGenerator struct {
-	sync.Mutex
+	sync.Mutex // Deprecated: mutex is internal and shall not be used.
+	mu         sync.Mutex
```

Comment:
Is there any reason for duplicating the fields?

```suggestion
	sync.Mutex // Deprecated: mutex is internal and shall not be used.
```

Reply:
Cannot you add a `//nolint` that this is required for backwards compatibility?